### PR TITLE
refactor(coordinator): organize provider protocol by message family

### PR DESCRIPTION
**Pull request by Gajesh2007**
The 1,121-line protocol message file mixed registration, capacity, inference, model commands, cache receipts and attestation with envelope decoding. Message records now live in files named for those operations within the same `protocol` package. `messages.go` keeps the type vocabulary and decoder in 221 lines; the package guide and protocol reference map each family to its source.

All JSON field order, tags, optional-field semantics, raw attestation/profile bytes, type constants and decoding branches are preserved. Existing tests remain beside the protocol package. Maintained source citations and review instructions follow the new files; historical records are unchanged.

Before:
```mermaid
flowchart LR
  Frame[Provider JSON frame] --> Decoder[messages.go: DecodeProviderMessage]
  Decoder --> Records[messages.go: registration, capacity, inference, model, cache and attestation records]
  Records --> Result[Typed payload or original decode error]
  Outbound[Coordinator typed message] --> Records
  Records --> JSON[JSON for the provider]
```

After:
```mermaid
flowchart LR
  Frame[Provider JSON frame] --> Decoder[messages.go: unchanged DecodeProviderMessage]
  Decoder --> Records[registration.go, heartbeat.go, backend_capacity.go, inference.go, models.go, prefix_cache.go, attestation.go, runtime_status.go]
  Records --> Result[Same typed payload or original decode error]
  Outbound[Coordinator typed message] --> Records
  Records --> JSON[Same JSON for the provider]
  Guide[doc.go and protocol reference] --> Records
```

Validation: all 267 protocol declarations, including their comments, compare exactly after relocation; 49 declarations move and four remain in `messages.go`. All existing test files are unchanged. Protocol race tests pass: 104 top-level / 253 including subtests, zero skips. All coordinator test packages compile; pinned lint reports zero issues; all 281 documentation checks pass. No wire migration or native provider/model run is required for this source-only organization. Stacked on #1038.

The current parent refresh preserves every protocol source and test byte from the reviewed change. The refreshed protocol race passes 104 top-level / 253 including subtests, zero skips; command build, lint and docs281 pass. The later release-review parent adds only its reviewed deletion fixture and documentation.

Review follow-up `7c93587de` widens the Swift protocol review prompt to the full `provider-swift/Sources/ProviderCore/Protocol/` owner. Parsed workflow configuration is unchanged outside that text; no serving source, gate or action changes.

### CODE REVIEW

**Comment by Gajesh2007** [comment]
@codex review

**Comment by Gajesh2007** [comment]
@codex review

File: .github/workflows/codex.yml [hunk 1]
```diff
@@ -93,7 +93,7 @@ jobs:
               git diff ${{ steps.meta.outputs.base_sha }}...${{ steps.meta.outputs.head_sha }}
 
             Focus on, in order of importance:
-            1. Correctness and regressions — broken imports, missing protocol symmetry between `provider-swift/Sources/ProviderCore/Protocol/Messages.swift` (Swift) and `coordinator/protocol/messages.go` (Go), release-bundle consistency across `.github/workflows/release-swift.yml` / `scripts/install.sh` / `LatestProviderVersion`, untested edge cases.
+            1. Correctness and regressions — broken imports, missing protocol symmetry between `provider-swift/Sources/ProviderCore/Protocol/Messages.swift` (Swift) and `coordinator/protocol/` (Go), release-bundle consistency across `.github/workflows/release-swift.yml` / `scripts/install.sh` / `LatestProviderVersion`, untested edge cases.
```
[the remark this answers is not in the document]
Comment (Gajesh2007) [reply .github/workflows/codex.yml]: Updated both active CLAUDE protocol guidance references and described the test as Go telemetry JSON-contract coverage. No Swift mirror execution is claimed. Fixed in a522834b1b. Documentation checks pass.

File: docs/developer/test.md [hunk 2]
```diff
@@ -66,6 +66,12 @@ make test   # coordinator-test prompt-sidecar-test provider-test ui-test benchma
 
 ### 2. Coordinator (Go)
 
+Wire-contract tests remain in `coordinator/protocol/` beside the message families.
+`GOTOOLCHAIN=go1.25.0 go test -race ./coordinator/protocol/...` checks the existing
+JSON round trips, omitted-versus-empty fields, Swift telemetry symmetry and
+provider-envelope/scanner equivalence cases. [The protocol source map](../reference/protocol-messages.md#source-files)
```
[the remark this answers is not in the document]
Comment (Gajesh2007) [reply docs/developer/test.md:72]: Updated both active CLAUDE protocol guidance references and described the test as Go telemetry JSON-contract coverage. No Swift mirror execution is claimed. Fixed in a522834b1b. Documentation checks pass.

**Comment by Gajesh2007** [comment]
@codex review

Updated both active CLAUDE protocol guidance references and described the test as Go telemetry JSON-contract coverage. No Swift mirror execution is claimed.

File: .github/workflows/codex.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Gajesh2007) [reply .github/workflows/codex.yml]: Fixed in `7c93587de`: the symmetry prompt now names the full `provider-swift/Sources/ProviderCore/Protocol/` directory, covering `Types.swift`, `Enums.swift`, and `Messages.swift`. Ruby/Psych validates both workflow versions and proves the parsed configuration is identical except for that one prompt replacement. Documentation check: 281 files passed.

**Comment by Gajesh2007** [comment]
@codex review

The Swift protocol symmetry prompt now covers the full Protocol/ owner. Parsed workflow configuration is unchanged outside that one prompt replacement; documentation check passes all 281 files.

**Comment by Gajesh2007** [comment]
Consolidated into #1056 for review and merge after #1094, #1095 and #1096. #1056 contains this branch's exact head (`7c93587dea2ed1a17b0bed1f180a7ba8cdd723b2`), including its reviewed fixes, with implementations and regressions reconciled into the coordinator subsystem owners. Closing the component PR to collapse the stack; its branch, commits and review history are retained. Follow #1056 for combined validation and merge status.
